### PR TITLE
Get Started / Data management - Make the the guide easier to understand and follow

### PR DESCRIPTION
**Pull request by omesser**
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

## background / motivation

Why is this important?
I've been having a lot of personal "raised eye-brow" moments following the get started guide myself. After some challenges that those things are "subjective" and variety of internal opinions, I "recruited" a couple of external "reviewers" - the gathered feedback from them going through get-started (only, one of them burnt out ofter data-management trail). the results were not good - lots of little questions, and "wtf moments", things that broke concentration. with one of them it took ~2h to go through data-management track. and not because there's a lot of content (which is also the case) - but because it's hard to follow to people who aren't already familiar with dvc and the docs themselves.

## goal
- goal: reduce "wtf moments". addressing how legible/clear and easy to follow the get-started guide is. this is driven by a list of issues brought up by 2 external reviewers (people I asked to review this and provide detail feedback). 1 is a sw eng, and 1 is not a sw eng (PM in tech startup). compounded by my experience reading this with "fresh eyes", and by past reports from users that found get-started is hard to follow (connection is not clear here, but I think the issues are probably similar).
- not a goal: make get-started shorter. This PR favors legibility and clarity over optimizing for less scrolling or something like that. I think this is a clear 1st order problem, while length is 2nd order, still, length is largely unchanged

> Note: moving data-pipelines to its own trail - not in this PR  (after this, I expect to do this in the next PR)

### dvc start feedback
- index - order unclear - set up dir before / after doing dvc init
- data-versioning
  - general: too many admons, details, and links - distracting, confusing whether we should jump around or follow linearly
  - can I run `dvc add` or need to expand to "get data"? (guide "linearity" is confusing)
  - mention of "placeholder" - what does it mean? (wording)
  - "⚠️ That didn't work!" - what does that mean. looks scary. why would the guide instruct me to run something that doesn't work? the actual working command (local remote) is hiding in an foldable (optional?) section.
  - configuring storage and pushing is mixed in an indistinguishable way. retrieving is separate. structure is perplexing
  - "Expand to delete locally cached data." and "Expand to delete locally cached data." - what? why? are those optional/needed? putting necessary steps in expandable details section is ambiguous. it's best to have more text

- Need consistency in using expandable sections ("details") - some are required, but seems like most are optional "peeks under the hood". inconsistent. Trying to make the guide shorter made it imposible to follow. Let's settle this 😄 - if you have to run it - don't hide it details section

## scope
- get started / data-management

👉 [deployment](https://dvc-org-get-started-cop-3jxkkl.herokuapp.com/doc/start) 👈

### CODE REVIEW

File: content/docs/start/data-management/data-pipelines.md
Comment (daavoo) [inline content/docs/start/data-management/data-pipelines.md:225]:
```suggestion
Finally, our `dvc.yaml` should have all 3 stages.
```

File: content/docs/start/data-management/data-pipelines.md
Comment (daavoo) [inline content/docs/start/data-management/data-pipelines.md:366]: Let's use an admon or just mention+link `dvc dag` at the top

**Review by daavoo — COMMENTED** [review]
Very minor comments.

I like the changes 🔥 . It makes it more of a reproducible tutorial, but I think it is okay. Feels like what I would follow to get started on data management.

Already said this but I felt it again when following this end-to-end: The metrics-parameters-plots page feels a bit off from the rest. 

It feels heavy and kind of gets into iterations and comparing which heavily overlaps with experiments.

File: content/docs/start/data-management/data-versioning.md
Comment (daavoo) [inline content/docs/start/data-management/data-versioning.md:193]:
```suggestion
#### 💡 Expand to simulate a "fresh pull"
```

File: content/docs/start/index.md
Comment (omesser) [reply content/docs/start/index.md:41]: full name, thanks for noticing 🙏

File: content/docs/start/data-management/data-versioning.md
Comment (omesser) [reply content/docs/start/data-management/data-versioning.md:70]: It's technically correct (example, no. 4 [here](https://englishpedia.net/inasentence/peek-in-a-sentence/))
But in terms of what would fit better here: "get a peek" or "take a peek" - not sure. I'm partial to either one. I think "see what happens under the hood" is not a 100% fit because it's losing the "fleeting nature" of a "peek" 😄

File: content/docs/start/data-management/data-versioning.md
Comment (omesser) [reply content/docs/start/data-management/data-versioning.md:232]: agree, a bit redundant, linked under `## Storing and sharing` actually.
A question - should we remove it from here, or you think we should keep this here and remove it from [line 98](https://github.com/iterative/dvc.org/pull/4508/files#diff-1b2d7fcff01951b9dfd24fb024b2fbd3073febaa65a64b61071315795ad1597dR98) and keep it here?
Reason I'm asking is I want the guide to not be too confusing throwing the user to read "extra stuff" when you just starting to learn about something. so I don't think we should necessarily make any possible reference a link to some section. it's overwhelming. if I read some text, run some commands, linearly, it's easier to do with less distractions and then have a similar admon guiding me to click the link if I want to read more.
Interested to get your opinion on this - remove it here, or there?

File: content/docs/start/data-management/data-versioning.md
Comment (omesser) [reply content/docs/start/data-management/data-versioning.md:287]: Agreed. done

File: content/docs/start/data-management/data-versioning.md
Comment (daavoo) [reply content/docs/start/data-management/data-versioning.md:232]: I would remove this admon at the bottom and change the admon at the top to something like:

```
DVC supports many remote storage types, including Amazon S3, NFS,SSH, Google Drive, Azure Blob Storage, and HDFS.

See more in the [Remote Storage Guide](https://dvc-org-get-started-cop-3jxkkl.herokuapp.com/doc/user-guide/data-management/remote-storage).
```

It fits better to me there, at the point where the user is setting a local remote.

File: content/docs/start/data-management/data-pipelines.md
Comment (daavoo) [reply content/docs/start/data-management/data-pipelines.md:366]: Sounds good, anything but not this pseudo-admon using `>`

File: content/docs/start/data-management/data-versioning.md
Comment (daavoo) [reply content/docs/start/data-management/data-versioning.md:70]: I see. 
I guess I was not familiar with the variation.

WDYT `take a quick look`?

Again, no strong opinion might be just me not being well versed in the jargon

**Comment by dberenbaum** [comment]
@omesser Thanks for the PR! From a quick look, it looks like it's all minor style changes, right? I'm fine for you to merge once @daavoo approves.

File: content/docs/start/data-management/data-versioning.md
Comment (omesser) [reply content/docs/start/data-management/data-versioning.md:70]: Also no strong opinion, just felt that this would "lighten the mood a bit".
@dberenbaum  - any preference? what would sound more natural?

**Review by daavoo — APPROVED** [review]

**Comment by omesser** [comment]
@daavoo  - thanks for the review 🙏 
I've addressed everything except the "metrics, parameters and plots" page - I've started changing it but I think it deserves its own PR to keep discussions contained and PRs manageable.
Objections to do this separately?

File: content/docs/start/data-management/data-versioning.md
Comment (omesser) [reply content/docs/start/data-management/data-versioning.md:70]: (if this is merged soon I'll followup on that in case we want to change, no objections 😄 )

**Comment by daavoo** [comment]
> Objections to do this separately?

Sounds good to me. I was just commenting on my feelings when following the full guide

**Comment by shcheklein** [comment]
Late to the party (didn't want to block a good PR). A few things that caught my attention right away (I didn't read it all tbh - surprising it's very quite hard to review when there are many minor changes):

1 . These statements:

Screenshot 2023-05-03 at 3 47 07 PM

I think they now stand out a bit too much to confuse people into thinking that DVC is _exacly_ Git for data, or Makefile (depends on what they see first). Since (I think) it's more of a nice too have thingy, I would drop them. Err on the side of not being confusing.

2. `Expand to get a peek under the hood` - don't have a strong opinion, but overall I would try to do less of these changes (we spend time reviewing this and even it's a bit better, my 2cs it's not worth it usually). Or at least would do these small stylistic changes in a separate PR. Again, primarily to save time and discussion when we review things. I'm not opposed to changing lang if needed. (one thing to keep in mind - in some (a lot in Get started) titles were crafter with SEO in mind - we spent some substantial amount of time analyzing keywords, etc, etc.

3. Moving `wget https://code.dvc.org/get-started/code.zi` before other commands (and similar changes). Intention was to focus on DVC commands first. It was secondary (with a bit more friction) priority to make it a tutorial. Now it feels way more like a tutorial. Same with creating virtualenv etc- it has become noisier (not bad for a tutorial, contradicts with the initial intention - show DVC first + give a way to run it for those who wants to run it).

4. `[content/docs/start/index.md](https://github.com/iterative/dvc.org/pull/4508/files#diff-1cd29580767fad4637f461df8460373ba6ec523dc598d7d3550657ed56cab101)` - here I would personally prefer to keep it shorter (e.g. when there are subtitles it's a sign that page has become too complicated already). Step by step with changes like this (adding a bit context) pages are getting longer (and we start seeing outside and inside a stream of tickets on simplifying things :) ).

**Comment by omesser** [comment]
Thanks for the inputs @shcheklein !
Sorry it took a few days to get back to this and follow up.

> I think they now stand out a bit too much to confuse people into thinking that DVC is exactly Git for data, or Makefile (depends on what they see first). Since (I think) it's more of a nice too have thingy, I would drop them. Err on the side of not being confusing.

I don't think anyone would be confused that this is "exactly like git for data". 
btw - _Think "Git for data"_ - was there before this PR even. 

I actually think it's a great idea. We do use these analogies in various scenarios (interviews, pitches, heard and used this multiple times), and some of our competitors are really doubling down on them as well:
Screenshot 2023-05-06 at 22 00 22

I don't think this causes any confusion tbh, on the contrary, it helps understand via analogy. I believe the get-started is exactly the place for these analogies as they help bridge the gap between the reader and the tool and foster understanding.
(I don't mind unbold of course 😄 , if you think it's too "loud" 👍 )
Let's please leave these in and get some feedback over time?

2. So far there isn't a lot of "drag" at all on these changes or PRs in general. cycle time is short, it's a side effort, and I'm coordinated with @dberenbaum and @daavoo for every "phase (=PR).

3 + 4. Please see the PR description! I made sure to explain motivation and give exact background. It is, btw, corroborated with more similar user inputs from @dberenbaum and @daavoo. I understand the original intent here behind optimizing for aesthetics/length and "reducing noise". However, we have plenty of data by now to understand it was optimizing for the wrong thing - and it got it into a state where new-comers are confused. This should be our no. 1 priority, over length or form (which I'm trying to respect and not hurt as well, as much as possible)

**Comment by shcheklein** [comment]
> I actually think it's a great idea. We do use these analogies in various scenarios (interviews, pitches, heard and used this multiple times), and some of our competitors are really doubling down on them as well:

Yes, it was me who put them there in the first place :) And even so, I think emphasizing them too much is harmful. It creates a wrong sense of understanding of what DVC is (simplification, narrowing, etc) and at the end hurts us. I've been fine with them being like "in this context you can think of DVC as ...", but making them an ad-like slogan is a bit too much (doesn't look like our usual style, sends the wrong message, etc, etc).

Yep, and I know about dolt and lake trying to optimize SEO for Git for data (Dolt btw is indeed a better described as Git for structured data, DVC is not actually Git for data, or at least not that extent). I'm neutral for that. I think people use it enough and we had it enough on the website. Over-optimizing for this would hurt us for the reason I described already. 

> So far there isn't a lot of "drag" at all on these changes or PRs in general

I saw a discussion. That alone is enough to me tbh to say that it's not worth it. It distracts from reviewing "actual" changes. Your call. My recommendation comes from many iterations with docs. If we want to do style changes, I would recommend doing them in a separate PR. It helps the team a lot usually.

> However, we have plenty of data by now to understand it was optimizing for the wrong thing

I don't think we have enough data on this (at least I don't have it). I have a lot of biased intuition, experience, etc based on all the iterations we had - tutorial style, short, long, etc. You _always_ have a group of users being confused by a particular format. I would not use voiced opinions, complains as data. Data should be coming from the actual usage (and there should be some baseline). In this particular case we won't see much difference. Each step like this makes it more and more complicated and at then I do believe we have a negative impact (ideally again should be confirmed by data which is not easy).

> I understand the original intent here behind optimizing for aesthetics/length and "reducing noise"

Original intent was not about noise, and especially not about aesthetics. Since I was driving this, I can share I guess a bit more. The original original intent was to have one single core DVC command on the first screen on each section. And ideally only that command (so that attention is focused on DVC, not other tools, so that people can as immediately as possible see and get the basic usage - they don't have to read ideally anything if possible)- that kind of reasoning, minimazing time to understading. Optimizing for a modern fast-paced of consuming thing - you have only 1-2 seconds of attention. 

Eventually, people added videos, kept growing explanations, moved under three long pages, vs short small self-contained things. We move forward and experiment- that's fine, but please let's keep fundamentals in mind (definitely, not aesthetics, nor length, nor noise).

**Pull request merged**
